app/scripts/services/membership/roleBindings.js

```diff
@@ -0,0 +1,149 @@
+      var binding = bindingTPL(roleName, namespace);
+      subject = qualifySubject(subject, namespace);
+      binding.subjects.push(angular.copy(subject));
+      return DataService.create('rolebindings', null, binding, context);
+    };
+
+    var addSubject = function(rb, subject, namespace, context) {
+      var tpl = bindingTPL();
+      var binding = _.extend(tpl, rb);
+      subject = qualifySubject(subject, namespace);
+      if(!subject) {
+        return binding;
+      }
+      if(_.isArray(binding.subjects)) {
+        if(_.includes(binding.subjects, subject)) {
+          return; // nothing to see here, folks.
+        }
+        binding.subjects.push(subject);
+      } else {
+        binding.subjects = [subject];
+      }
+      cleanBinding(binding);
+      return DataService.update('rolebindings', binding.metadata.name, binding, context);
+    };
+
+    // has to handle multiple bindings or multiple reference to a subject within a single binding
+    var removeSubject = function(subjectName, role, roleBindings, context) {
+      var matches = _.filter(roleBindings, function(roleBinding) {
+        return _.isEqual(roleBinding.roleRef.name, role);
+      });
+      return $q.all(
+                _.map(matches, function(binding) {
+                  var tpl = bindingTPL();
+                  binding = _.extend(tpl, binding);
+                  cleanBinding(binding);
+                  binding.subjects = _.reject(binding.subjects, function(subject) {
+                    return _.isEqual(subject.name, subjectName);
+                  });
+                  return _.isEqual(_.size(binding.subjects), 0) ?
+                          DataService.delete('rolebindings', binding.metadata.name, context) :
+                          DataService.update('rolebindings', binding.metadata.name, binding, context);
+                }));
+    };
+
+    var findFirstByRole = function(role, roleBindings) {
+      return _.find(roleBindings, function(binding) {
+        return _.isEqual(binding.roleRef.name, role);
+      });
+    };
+
+    var get = function(name, context, opts) {
+      return DataService.get('rolebindings', name, context, opts);
+    };
+
+    var list = function(context, fn, opts) {
+      return DataService
+              .list('rolebindings', context, function(resp) {
+                cache = resp.by('metadata.name');
+                fn(resp);
+              }, opts);
+    };
+
+    // since rolebindings are a generated from policybindings, they cannot be
+    // watched directly. But we can watch the policybindings and then request
+    // a regenerated rolebinding list
+    var watch = function(context, callbackFn, opts) {
+      return DataService
+              .watch('policybindings', context, function() {
+                return list(context, callbackFn, opts);
+              });
+    };
+
+    return {
+      get: get,
+      list: list,
+      watch: watch,
```

Comment:
I don't think I like this pattern of wrapping get/list/watch again.  You are going to have to convince me, cc @spadgett